src/coreclr/jit/codegenxarch.cpp

```diff
@@ -2584,6 +2572,7 @@ void CodeGen::genLclHeap(GenTree* tree)
             }
             else
             {
+                assert(tree->AvailableTempRegCount() == 1);
                 regCnt = tree->ExtractTempReg();
```

Comment:
If there's only one temp reg expected, then use GetSingleTempReg() which already does this assert.
```suggestion
                regCnt = tree->GetSingleTempReg();
```

Reply:
Ah, that's good and convenient.